src/method-management/method-management.ts

```diff
@@ -39,24 +38,56 @@ export const createMfaMethodHandler = async (
     } = {},
   } = JSON.parse(event.body || "{}");
   try {
-    assert(email, "no email provided");
-    assert(otp, "no otp provided");
-    assert(credential, "no credential provided");
-    assert.match(
-      priorityIdentifier,
-      /^(PRIMARY|SECONDARY)$/,
-      "invalid priorityIdentifier"
+    validateFields(
+      { email, otp, credential, priorityIdentifier, mfaMethodType },
+      {
+        priorityIdentifier: /^(PRIMARY|SECONDARY)$/,
+        mfaMethodType: /^(AUTH_APP|SMS)$/,
+      },
     );
-    assert.match(mfaMethodType, /^(AUTH_APP|SMS)$/, "invalid mfaMethodType");
   } catch (e) {
-    return {
-      statusCode: 400,
-      body: JSON.stringify({ error: (e as Error).message }),
-    };
+    return formatResponse(400, { error: (e as Error).message });
   }
 
-  return {
-    statusCode: 200,
-    body: JSON.stringify({}),
-  };
+  return formatResponse(200, {});
+};
+
+export const updateMfaMethodHandler = async (
+  event: APIGatewayProxyEvent,
+): Promise<APIGatewayProxyResult> => {
+  try {
+    assert(event.body, "no body provided");
+    const body = JSON.parse(event.body);
+    const mfaIdentifier = event.pathParameters?.mfaIdentifier;
+    const {
+      email,
+      credential,
+      otp,
+      mfaMethod: {
+        priorityIdentifier = undefined,
```

Comment:
Hi Alfonso, should this nmot include the mfaIndetifier as well?

Reply:
See here:  https://github.com/govuk-one-login/openapi-specs/blob/3722748aa2ea81f05c68f899939031cbb147f6d7/auth/method-management-api.yaml#L168

Reply:
Thought it should accept something like this:

{
  "email": "testuser@testexampledomain.co.uk",
  "credential": "SMS",
  "otp": "123456",
  "mfaMethod": {
    "mfaIdentifier": 1213,
    "priorityIdentifier": "SECONDARY",
    "mfaMethodType": "SMS",
    "endPoint": "PHONE",
    "methodVerfied": true
  }
}

Reply:
@peterfajemisincabinetoffice well spotted. I messaged the auth team and they agree there is a discrepancy and sending the identifier in the body is not needed. They are going to create a PR for it. Once is done and merged, I'll review this PR to make sure it's aligned with the open api spec.